docs/content/release-notes/flink-1.17.md

```diff
@@ -0,0 +1,201 @@
+The Datadog reporter now adds a "flink." prefix to metric identifiers if "useLogicalIdentifier" is 
+enabled. This is required for these metrics to be recognized as Flink metrics, not custom ones.
+
+#### Use separate Prometheus CollectorRegistries
+##### [FLINK-30020](https://issues.apache.org/jira/browse/FLINK-30020)
+The PrometheusReporters now use a separate CollectorRegistry for each reporter instance instead of 
+the singleton default registry. This generally shouldn't impact setups, but it may break code that 
+indirectly interacted with the reporter via the singleton instance (e.g., a test trying to assert 
+what metrics are reported).
+
+### Checkpoints
+
+#### Drop TypeSerializerConfigSnapshot and savepoint support from Flink versions < 1.8.0
+##### [FLINK-29807](https://issues.apache.org/jira/browse/FLINK-29807)
+Savepoints using `TypeSerializerConfigSnapshot` are no longer supported. That means, all savepoints 
+from Flink < 1.8.0 are no longer supported. Furthermore, savepoints from Flink < 1.17.0 created 
+with custom serializer using deprecated since Flink 1.8.0 class `TypeSerializerConfigSnapshot` are 
+also no longer supported.
+
+If you are only using built-in serializers (Pojo, Kryo, Avro, 
+Tuple, ...), and your savepoint is from 1.8.0 <= Flink, you don't have to do anything.
+
+If you are only using built-in serializers (Pojo, Kryo, Avro, Tuple, ...), and your savepoint is 
+from Flink < 1.8.0, please first upgrade your job to 1.8.x <= Flink <= 1.16.x, before upgrading in 
+the second step to Flink >= 1.17.x.
+
+If previously you were using a custom serialiser that 
+depends on `TypeSerializerConfigSnapshot`, please first while still using 1.8.x <= Flink <= 1.16.x 
+upgrade your serializer to `TypeSerializerSnapshot`, take a savepoint and restore from that 
+savepoint in Flink >= 1.7.0.
+
+### Python
+
+#### Remove duplicated operators in the StreamGraph for Python DataStream API jobs
+##### [FLINK-31272](https://issues.apache.org/jira/browse/FLINK-31272)
+This duplicated operator issue has been addressed since 1.15.4, 1.16.2 and 1.17.0. For jobs which 
+are not affected by this issue, there are no backward compatibility issues. However, for jobs which 
+are affected, it may not be possible to restore from savepoints generated from versions 
+1.15.0 ~ 1.15.3 and 1.16.0 ~ 1.16.1.
+
+
+### Dependency upgrades
+
+#### Upgrade FRocksDB to 6.20.3-ververica-2.0
+##### [FLINK-30836](https://issues.apache.org/jira/browse/FLINK-30836)
+Upgrade FRocksDB to 6.20.3-ververica-2.0.
+
+#### Upgrade the minimal supported hadoop version to 2.10.2
+##### [FLINK-29710](https://issues.apache.org/jira/browse/FLINK-29710)
+The minimum Hadoop version supported by Apache Flink has been updated to version 2.10.2. For Hadoop 
+3, the minimum Hadoop version that is now supported is version 3.2.3.
+
+#### Upgrade Calcite version to 1.27
+##### [FLINK-20873](https://issues.apache.org/jira/browse/FLINK-20873)
```

Comment:
Since finally in Flink 1.17.0 it was upgraded to 1.29.0, I think it would make sense to use the latest Calcite version in 1.17 branch
```suggestion
#### Upgrade Calcite version to 1.29.0
##### [FLINK-20873](https://issues.apache.org/jira/browse/FLINK-20873)
##### [FLINK-21239](https://issues.apache.org/jira/browse/FLINK-21239)
##### [FLINK-29932](https://issues.apache.org/jira/browse/FLINK-29932)
Calcite upgrade brings optimization changing logical plans for some queries involving `Sarg` and 
queries with `count` of non-distinct values.
```